### PR TITLE
Fix Prefast C6101 on Foundation Solution 

### PR DESCRIPTION
**Pull request by kythant**
A microsoft employee must use /azp run to validate using the pipelines below.

WARNING:
Comments made by azure-pipelines bot maybe inaccurate.
Please see pipeline link to verify that the build is being ran.

For status checks on the main branch, please use TransportPackage-Foundation-PR
(https://microsoft.visualstudio.com/ProjectReunion/_build?definitionId=81063&_a=summary)
and run the build against your PR branch with the default parameters.

### CODE REVIEW

File: dev/UndockedRegFreeWinRT/typeresolution.cpp [hunk 1]
```diff
@@ -213,8 +221,19 @@ namespace UndockedRegFreeWinRT
...
+            *ppMetaDataImport = nullptr;
+        }
+        if (pmdTypeDef != nullptr)
+        {
+            *pmdTypeDef = 0;
+        }
+
         wchar_t szCandidateFileName[MAX_PATH + 1]{};
         HRESULT hr{ StringCchCopy(szCandidateFileName, ARRAYSIZE(szCandidateFileName), pszFullName) };
+        hr = WindowsCreateString(L"", 0, phstrMetaDataFilePath);
```
Comment (alexlamtest) [inline dev/UndockedRegFreeWinRT/typeresolution.cpp:235]: Is it intentional that we now overwrite the hr result from the stringCchCopy() w/o first checking for error?

Comment (kythant) [reply dev/UndockedRegFreeWinRT/typeresolution.cpp:235]: oh good point

**Pull request merged**